### PR TITLE
feat: Replace version map with distro-info

### PR DESCRIPTION
**Pull request by TheRealFalcon**
Every 6 months we add another entry to
UBUNTU_RELEASE_VERSION_MAP. Instead, call distro-info to be more future proof.

### CODE REVIEW

**Review by blackboxsw — APPROVED** [review]
Minor change, then +1

File: pycloudlib/util.py [hunk 1]
```diff
@@ -20,22 +20,40 @@
...
+    >>> release_version_from_series("noble")
+    "24.04"
+    """
+    try:
+        return subprocess.run(
+            ["distro-info", "--release", f"--series={series}"],
+            check=True,
+            capture_output=True,
+            universal_newlines=True,
+        ).stdout.strip()
```
Comment (blackboxsw) [inline pycloudlib/util.py:50]: This returns `24.04 LTS` on noble and other LTS releases. I think we want to trim that content too as that space will give us problems in instance names and the extra lengths of instance names gives us probs on lxc integration tests in pro I believe.
```suggestion
        return subprocess.run(
            ["distro-info", "--release", f"--series={series}"],
            check=True,
            capture_output=True,
            universal_newlines=True,
        ).stdout.split()[0]
```

Comment (TheRealFalcon) [reply pycloudlib/util.py]: Thanks!

**Review by holmanb — CHANGES_REQUESTED** [review]
> Every 6 months we add another entry to UBUNTU_RELEASE_VERSION_MAP. Instead, call distro-info to be more future proof.

This is an Ubuntu-only dependency. I like to run integration tests on non-Ubuntu hosts sometimes.

-1 from me

**Comment by TheRealFalcon** [comment]
@holmanb 

> This is an Ubuntu-only dependency

I'm replacing one Ubuntu-only dependency with another Ubuntu-only dependency. Image lookups have always been Ubuntu-only. That didn't prevent anybody from specifying their own image id. Neither will this.

**Comment by holmanb** [comment]
>> This is an Ubuntu-only dependency
>
> I'm replacing one Ubuntu-only dependency with another Ubuntu-only dependency. Image lookups have always been Ubuntu-only. That didn't prevent anybody from specifying their own image id. Neither will this.

You are describing launching non-Ubuntu images, and yes currently we only support launching Ubuntu via these lookups. That isn't what breaks by this change.

As I said before, I like to run our integration tests (which use pycloudlib) **on non-Ubuntu hosts** sometimes. Requiring an Ubuntu-only dependency will mean that now if one wants to run the integration test suite, they will not be able to launch it from a Debian / Arch / Redhat system.

**Comment by TheRealFalcon** [comment]
> they will not be able to launch it from a Debian / Arch / Redhat system

What about this code is preventing that?

**Comment by TheRealFalcon** [comment]
Oh, I see...specifically looking up Ubuntu from non-Ubuntu.

**Comment by holmanb** [comment]
@TheRealFalcon I'm experiencing deja-vu

https://github.com/canonical/cloud-init/pull/4586

**Pull request closed without merging**